src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs

```diff
@@ -307,9 +306,7 @@ await GetPackageMetadataAsync(packageId.ToString(), packageVersion, packagesSour
             return (source, packageVersion);
         }
 
-        private string GetNupkgUrl(string baseUri, PackageId id, NuGetVersion version) =>
-            baseUri + id.ToString() + "/" + version.ToNormalizedString() + "/" + id.ToString() +
-            "." + version.ToNormalizedString() + ".nupkg";
+        private string GetNupkgUrl(string baseUri, PackageId id, NuGetVersion version) => new VersionFolderPathResolver(baseUri).GetPackageFilePath(id.ToString(), version);
```

Comment:
The nupkgs are in the folders for id/version, so this will miss out on that and generate an incorrect link. 

Per: https://[api.nuget.org/v3/index.json](https://api.nuget.org/v3/index.json), 
```json
     {
            "@id": "https://api.nuget.org/v3-flatcontainer/",
            "@type": "PackageBaseAddress/3.0.0",
            "comment": "Base URL of where NuGet packages are stored, in the format https://api.nuget.org/v3-flatcontainer/{id-lower}/{version-lower}/{id-lower}.{version-lower}.nupkg"
        },```
        
Eventually FindPackagesByIdResource could be used to just "solve all of that for you.

Reply:
You probably want a test for this.

Reply:
Thank you @nkolev92  for taking a look at these changes and noticing this. 

It looks like `FindPackagesByIdResource` is not available yet: https://github.com/search?q=org%3ANuGet%20FindPackagesByIdResource%20%20&type=code

Is this correct? In this case, I suppose we would just use the old code but call the to lower case normalization function on the version string? Or we could call GetPackageFilePath twice and edit it a bit to match the existing definition?

Reply:
It's available: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Protocol/Resources/FindPackageByIdResource.cs. 

You need to do `await sourceRepository.GetResourceAsync<FindPackagesByIdResource>()`

Reply:
> Is this correct? In this case, I suppose we would just use the old code but call the to lower case normalization function on the version string?

Yeah, you can do either, probably easier to read if you're explicit with the old code.
Especially if you're going to refactor to the FindPackagesByIdResource later.

Reply:
Ok, I will revert to the previous code and annotate on the issue for refactoring later, thanks!